### PR TITLE
FirstClientTestFixes

### PR DESCRIPTION
**Pull request by ShannyWorkiz**

### CODE REVIEW

File: cypress/page-objects/Clients/createClientPage.ts [hunk 1]
```diff
@@ -10,17 +10,21 @@ export class CreateClientPage {
         let firstName: string = RandomFunctions.generateRandomString(8);
         cy.get(`${CreateClientPage.firstNameLocator}`).type(firstName);
         cy.get('[name="address_serach"]').type(Constans.ADDRESS);
+        cy.wait(1000);
+        cy.get('[name="address_serach"]').type(' r');
```
Comment (IdoAdler2) [inline cypress/page-objects/Clients/createClientPage.ts:14]: @ShannyWorkiz change to {delay:200} inside the type(address)

File: cypress/page-objects/Clients/createClientPage.ts [hunk 2]
```diff
@@ -10,17 +10,21 @@ export class CreateClientPage {
...
+        })
     }
 
-    save():ClientPage {
+    save(clientIdAlias: string):ClientPage {
         cy.get('button').contains('Save').click();
-        cy.url().should('contain', 'client/')
-        return new ClientPage;
+        cy.url().should('contain', 'client/');
+        return new ClientPage(clientIdAlias);
```
Comment (IdoAdler2) [inline cypress/page-objects/Clients/createClientPage.ts:27]: @ShannyWorkiz it's better if the save button will not get param.
Let's make the 'save' function void, and we will return the clientpage from outside. from the function that called the save.

File: cypress/tests/registrations/clients.spec.ts [hunk 3]
```diff
@@ -15,8 +16,10 @@ describe('Clients tests', () => {
 
     it.only('When creating new client it appears in the clients report',() =>{
         let allClientsPage = pageRouter.goToClientsPage();
-        let clientPage = allClientsPage.createClient();
-        cy.get('@clientId').then((clientId) =>{
+        let clientAliasName = RandomFunctions.generateRandomString(5);
+        allClientsPage.createClient(clientAliasName);
+        let alias = '\@' +`${clientAliasName}`;
```
Comment (IdoAdler2) [inline cypress/tests/registrations/clients.spec.ts:21]: @ShannyWorkiz lets move this code: ''\@' +`${clientAliasName}`' inside a function in rundom functions called generate alias();

File: cypress/tests/registrations/clients.spec.ts [hunk 4]
```diff
@@ -15,8 +16,10 @@ describe('Clients tests', () => {
 
     it.only('When creating new client it appears in the clients report',() =>{
         let allClientsPage = pageRouter.goToClientsPage();
-        let clientPage = allClientsPage.createClient();
-        cy.get('@clientId').then((clientId) =>{
+        let clientAliasName = RandomFunctions.generateRandomString(5);
```
Comment (IdoAdler2) [inline cypress/tests/registrations/clients.spec.ts:19]: @ShannyWorkiz lets create function in randomFunctions called, 'generateAlias' that does 2 actions:
1. ${aliasName} = this.generateRandomString(5);
2. return  '\@' +`${aliasName}`;

File: cypress/page-objects/Clients/createClientPage.ts (hunk 2 above)
Comment (ShannyWorkiz) [reply cypress/page-objects/Clients/createClientPage.ts:27]: We can't - it returns a client page. and we pass an alias so we could identify a client when we have more than one, this was our solution, remember?

File: cypress/tests/registrations/clients.spec.ts (hunk 4 above)
Comment (ShannyWorkiz) [reply cypress/tests/registrations/clients.spec.ts:19]: You can't - the @ Should only be used when calling to an alias - not when creating one.

File: cypress/tests/registrations/clients.spec.ts (hunk 3 above)
Comment (ShannyWorkiz) [reply cypress/tests/registrations/clients.spec.ts:21]: Fixed.

File: cypress/page-objects/Clients/createClientPage.ts (hunk 1 above)
Comment (ShannyWorkiz) [reply cypress/page-objects/Clients/createClientPage.ts]: @IdoAdler2 fixed

File: cypress/page-objects/Clients/createClientPage.ts (hunk 2 above)
Comment (ShannyWorkiz) [reply cypress/page-objects/Clients/createClientPage.ts]: @IdoAdler2 Fixed Changed to counter in client class

File: cypress/tests/registrations/clients.spec.ts (hunk 4 above)
Comment (ShannyWorkiz) [reply cypress/tests/registrations/clients.spec.ts]: @IdoAdler2 Fixed Changed to counter in client class

**Review by IdoAdler2 — APPROVED** [review]

**Pull request merged**